ext/guards.go

```diff
@@ -246,3 +247,55 @@ func callInStrStrStrIntOutStr(fn func(string, string, string, int64) (string, er
 		return types.String(out)
 	}
 }
+
+func callInListStrOutStr(fn func([]string) (string, error)) functions.UnaryOp {
+	return func(args1 ref.Val) ref.Val {
+		vVal, ok := args1.(traits.Lister)
+		if !ok {
+			return types.MaybeNoSuchOverloadErr(args1)
+		}
+		strings := make([]string, vVal.Size().Value().(int64))
+		i := 0
+		for it := vVal.Iterator(); it.HasNext() == types.True; {
+			v, ok := it.Next().(types.String)
+			if !ok {
+				return types.MaybeNoSuchOverloadErr(v)
```

Comment:
You'll want to store a reference to the `it.Next()` result and pass that to the `MaybeNoSuchOverloadErr()` call here.